src/components/FeedbackForm.jsx

```diff
@@ -0,0 +1,185 @@
+import axios from 'axios';
+import { useState } from 'react';
+
+export default function FeedbackForm() {
+  const [submitted, setSubmitted] = useState(false);
+  const [valid, setValidity] = useState(false);
+  const [formValues, setFormValues] = useState({
+    firstName: '',
+    lastName: '',
+    mail: '',
+    rating: '',
+    feedback: ''
+  });
+
+  const changeVal = e => {
+    setFormValues({ ...formValues, [e.target.id]: e.target.value });
+  };
+
+  const radioSelected = e => {
+    setFormValues({ ...formValues, rating: e.target.value });
+  };
+
+  const resetForm = data => {
+    if (data.status === 200) {
+      setFormValues({
+        firstName: '',
+        lastName: '',
+        mail: '',
+        rating: '',
+        feedback: ''
+      });
+      setSubmitted(false);
+      setValidity(false);
+      Array.from(document.getElementsByName('rating')).map(item => (item.checked = false));
+    } else {
+      alert('Something went wrong! Please try again.');
+    }
+  };
+
+  const formSubmit = e => {
+    e.preventDefault();
+    setSubmitted(true);
+    const { firstName, lastName, mail, rating, feedback } = formValues;
+    if (firstName && lastName && mail && rating && feedback) {
+      setValidity(true);
+      setTimeout(function () {
+        localStorage.setItem('survey-form', JSON.stringify(formValues));
+        resetForm({ status: 200 });
+      }, 5000);
```

Comment:
why do you need timeout while saving in local storage?

Reply:
to see the success message before reset

Reply:
please rewrite your code to not use setTimeout, it's redundant

Reply:
any suggestions how to do it?

Reply:
check again logic of showing success message and update it. 
usually success message can be shown after form successfully submitted and disappear after some timeout (but it definitely should not depend on saving form process, in your case form should be saved immediately) or it can be left until user clicks submit button again